cli/src/invite_device.rs

```diff
@@ -0,0 +1,118 @@
+
+    if let Some(device_slughash) = invite_device.device {
+        let mut possible_devices = vec![];
+
+        for device in &devices {
+            if device.slughash().starts_with(&device_slughash) {
+                possible_devices.push(device);
+            }
+        }
+
+        match possible_devices.len() {
+            0 => {
+                println!("Device `{device_slughash}` not found, available devices:");
+                format_devices(&devices);
+            }
+            2 => {
+                println!("Multiple devices found for `{device_slughash}`:");
+                format_devices(&devices);
+            }
+            _ => {
+                let device = &possible_devices[0];
+
+                let device = match device.ty {
+                    DeviceFileType::Password => {
+                        let password = rpassword::prompt_password("password:")?.into();
+                        let access = DeviceAccessStrategy::Password {
+                            key_file: device.key_file_path.clone(),
+                            password,
+                        };
+
+                        // The password is invalid or the binary is compiled with fast crypto
+                        load_device(&config_dir, &access).await?
+                    }
+                    DeviceFileType::Smartcard => {
+                        let access = DeviceAccessStrategy::Smartcard {
+                            key_file: device.key_file_path.clone(),
+                        };
+
+                        load_device(&config_dir, &access).await?
+                    }
+                    DeviceFileType::Recovery => {
+                        return Err(anyhow::anyhow!(
+                            "Unsupported device file authentication `{:?}`",
+                            device.ty
+                        ));
+                    }
+                };
+
+                let cmds = AuthenticatedCmds::new(
+                    &config_dir,
+                    device.clone(),
+                    ProxyConfig::new_from_env()?,
+                )?;
+
+                let handle = SpinnerBuilder::new()
+                    .spinner(&DOTS)
+                    .text("Creating device invitation")
+                    .start();
+
+                let rep = cmds
+                    .send(invite_new::Req(UserOrDevice::Device { send_email: false }))
+                    .await?;
+
+                let url = match rep {
+                    InviteNewRep::Ok { token, .. } => BackendInvitationAddr::new(
+                        device.organization_addr.clone(),
+                        device.organization_id().clone(),
+                        InvitationType::Device,
+                        token,
+                    )
+                    .to_url(),
+                    rep => {
+                        return Err(anyhow::anyhow!(
+                            "Backend refused to create device invitation: {rep:?}"
```

Comment:
We prefer "server" to "backend" for user-facing texts (docs, cli, gui, ...)
```suggestion
                            "Server refused to create device invitation: {rep:?}"
```